### PR TITLE
FAD-2847: Generate static styled API documentation

### PR DESCRIPTION
**Pull request by ewandennis**
This branch generates static API docs from files created in sparkpost-api-documentation-DEV#FAD-2847.  The README has details on the workflow.

NOTE: This branch also contains:

 - fixes for some Jekyll / Liquid base URL config variable usage to allow deployment as a top-level site (e.g. developers.sparkpost.com) or a sub-site (e.g. bob.github.io/devhubstaging/)
 - A top nav change to collapse into hamburger _before_ the left and right nav items wrap over 2 lines

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +25,34 @@ The following must be installed in order run Jekyll locally.
...
+
+For an easy dev setup:
+
+- Check out sparkpost-api-documentation-DEV
+- Check out sparkpost.github.io alongside
+- Start the API docs watch-and-build process:
+
+```bash
+cd sparkpost-api-documentation-DEV
+grunt staticDev
```
Comment (richleland) [inline README.md]: This flow works, but might not be super simple for devs that are used to the existing setup. As discussed on Slack, maybe we can have this command check out the dev hub? Though the dev hub does require ruby and bundler to work. Open to other options.

Comment (richleland) [inline README.md]: I wonder if we could just make the template a command line flag - use the default aglio templates for local dev/serving and use the custom templates for the build. Thoughts?

File: _config.yml [hunk 2]
```diff
@@ -8,7 +8,7 @@ description: >
   developers. Our developer hub is a collection of resources to help you succeed
   with SparkPost. What will you build?
 baseurl: "" # the subpath of your site, e.g. /blog/
-url: "https://developers.sparkpost.com" # the base hostname & protocol for your site
+url: "https://sparkpost.github.io" # the base hostname & protocol for your site
```
Comment (richleland) [inline _config.yml]: Why are we changing the URL?

File: _sass/_layout.scss [hunk 3]
```diff
@@ -159,3 +159,238 @@ body {
...
+// --------------------------
+
+.page-api-reference {
+  color: $c-gray-darker;
+
+  /*
+   * <TABLE> becomes <TABLE class="table">
+   * .table is the basic Bootstrap table layout
+   */
+  TABLE {
```
Comment (richleland) [inline _sass/_layout.scss]: why the upper-case element name?

File: _sass/_layout.scss [hunk 4]
```diff
@@ -159,3 +159,238 @@ body {
...
+     * Top margin to avoid site-header
+     */
+    #contentblock {
+      margin-left: 25%;
+      margin-top: 76px;
+    }
+  } /* @media */
+
+  .site-header {
+    NAV .container {
```
Comment (richleland) [inline _sass/_layout.scss]: upper-case?

File: _sass/_layout.scss [hunk 5]
```diff
@@ -159,3 +159,238 @@ body {
...
+      font-size: 9pt;
+    }
+
+    .markdown, .resource {
+      /*
+       * Wrap code to avoid huge overflow into the right margin.
+       * The metrics API endpoints are worst for this: lots of query params.
+       * Taken from https://css-tricks.com/snippets/css/make-pre-text-wrap/
+       */
+      CODE {
```
Comment (richleland) [inline _sass/_layout.scss]: upper-case?

File: _sass/_layout.scss [hunk 6]
```diff
@@ -159,3 +159,238 @@ body {
...
+        word-wrap: break-word;
+      }
+    }
+
+    @import "_api-syntax-highlighting";
+
+  } // #contentblock
+
+  nav#apimenu {
```
Comment (richleland) [inline _sass/_layout.scss]: `nav` is unnecessary here because we have an ID

File: README.md (hunk 1 above)
Comment (ewandennis) [inline README.md]: This discussion is continued in the [sparkpost-api-documentation-DEV PR](https://github.com/SparkPost/sparkpost-api-documentation-DEV/pull/143).  tl;dr we might want to decouple production rendering through Jekyll from dev-time renders which use just the API docs tooling.

File: _config.yml (hunk 2 above)
Comment (ewandennis) [inline _config.yml]: Bad.  Fixed.

File: _sass/_layout.scss (hunk 3 above)
Comment (ewandennis) [inline _sass/_layout.scss]: Old habit.  Fixed in each case below.

File: _sass/_layout.scss (hunk 6 above)
Comment (ewandennis) [inline _sass/_layout.scss]: Indeed.  CSS evolutionary throwback.  Removed.

**Comment by richleland** [comment]
Build is failing, otherwise LGTM.

**Comment by ewandennis** [comment]
The failing build is a broken link in 2 API docs.  Interestingly its also broken in the published Apiary docs but we don't link check those before publication.

**Pull request merged**